13.0 - [REF] account: Adding unittests for concurrency issues in account_move sequences

moylop260 opened the pull request:
Reproducing all the new cases of concurrency for Odoo v14.0 account.move records but for v13.0 in the following cases:
- Creating draft invoices
- Creating payments
- Deadlock payment vs invoice
- Editing last invoice and creating new one
- Editing last payment and creating new one
- Reconciling last invoice and creating new one
- Reconciling last payment and creating new one
- No be able to configure to standard sequences anymore

Related to v14.0 account move sequence concurrency issues:
 - https://github.com/odoo/odoo/pull/91525
 - https://github.com/odoo/odoo/issues/90465

Issue describing this:
 - https://github.com/odoo/odoo/issues/91873

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/91614).

moylop260 commented:
Fixed for >=v16.0 from https://github.com/odoo/odoo/pull/104606
Fixed for >=14.0 and <16.0 from OCA module [account_move_name_sequence](https://github.com/OCA/account-financial-tools/tree/14.0/account_move_name_sequence)
